### PR TITLE
Extend `map_clone` lint to also work on non-explicit closures

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I found it weird that this case was not handled by the current line so I added it. The only thing is that I don't see an obvious way to infer the current type to determine if it's copyable or not, so for now I always suggest `cloned` and I added a FIXME.

r? @llogiq 

changelog: Extend `map_clone` lint to also work on non-explicit closures

### CODE REVIEW

**Comment by llogiq** [comment]
Would it then make sense to also change the message because `Clone::clone` is not a closure.

Perhaps something like "manually implementing `.cloned()`".

**Comment by GuillaumeGomez** [comment]
Sounds good to me! Just realized that I also left some code I was working for another update... I also need to fix dogfood which founds issues thanks to this update.

**Comment by GuillaumeGomez** [comment]
Fixed the CI failures. :)

File: clippy_lints/src/methods/map_clone.rs [hunk 1]
```diff
@@ -19,50 +19,63 @@ pub(super) fn check(cx: &LateContext<'_>, e: &hir::Expr<'_>, recv: &hir::Expr<'_
...
                     },
                     _ => {},
                 }
             },
+            hir::ExprKind::Path(qpath) => {
+                if let Some(path_def_id) = cx.qpath_res(&qpath, arg.hir_id).opt_def_id()
+                    && match_def_path(cx, path_def_id, &paths::CLONE_TRAIT_METHOD)
+                {
+                    // FIXME: It would be better to infer the type to check if it's copyable or not
+                    // to suggest to use `.copied()` instead of `.cloned()` where applicable.
```
Comment (llogiq) [inline clippy_lints/src/methods/map_clone.rs:75]: Would getting the iterator type from the `map(_)` and then extracting the `Item` type work here?

**Review by llogiq — APPROVED** [review]

File: clippy_lints/src/methods/map_clone.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply clippy_lints/src/methods/map_clone.rs:75]: Worth a try.

**Comment by llogiq** [comment]
Looks good to me. Any further extensions can be a follow up PR.

@bors r+

**Pull request merged**